.github/workflows/build_and_test.yml

```diff
@@ -1092,12 +1093,47 @@ jobs:
               }
 
               console.log(`Release for tag ${tag} does not exist. Creating it...`);
+
+              // Get the merge commit message to extract the PR number
+              const commit = await github.rest.git.getCommit({
+                owner: context.repo.owner,
+                repo: context.repo.repo,
+                commit_sha: context.sha,
+              });
+              const prMatch = commit.data.message.match(/Merge pull request #(\d+)/);
+              const prNumber = prMatch ? prMatch[1] : null;
```

Comment:
This looks a bit fragile, should we try something to find PR number using a commit hash? 
Like this: https://github.com/neondatabase/neon/blob/b0b4b7dd8f0dc6a6e780b7ed880b26f70d9016a1/.github/workflows/build_and_test.yml#L1171-L1177

Reply:
might make more sense, yeah

Reply:
will do tomorow

Reply:
yep, that feels more robust. Tested in https://github.com/neondatabase-labs/test-githubscript-thingy/actions/runs/13009939390, works as expected.